src/libsyntax/parse/literal.rs

```diff
@@ -263,8 +263,8 @@ impl Lit {
     /// Losslessly convert an AST literal into a token stream.
     crate fn tokens(&self) -> TokenStream {
         let token = match self.token {
-            token::Bool(symbol) => Token::Ident(Ident::with_empty_ctxt(symbol), false),
-            token => Token::Literal(token, self.suffix),
+            token::Bool(symbol) => token::Ident(Ident::new(symbol, self.span), false),
+            token => token::Literal(token, self.suffix),
         };
         TokenTree::Token(self.span, token).into()
     }
```

Comment:
Don't we want a test for this? I imagine testing booleans in particular  will be a hassle, but perhaps we can write/find  single test which checks all the literals (by, for example, debug-dumping spans in proc macro)

Reply:
So I tried to reproduce this empty span pretty hard without success, until I noticed that the correct span is placed into `TokenTree::Token` a few lines below.
Apparently, the token span is actually used in most (all?) places and the span inside of the ident (which should be its duplicate) is not used.

Perhaps `token::Ident` should use `Symbol` to avoid confusion and potential bugs (added to the TODO list).